jujugui/static/gui/src/app/extensions/autodeploy-extension.js

```diff
@@ -47,7 +47,10 @@ YUI.add('autodeploy-extension', function(Y) {
           env = this.get('env') || this.env,
           unplacedUnits = db.units.filterByMachine(null);
       unplacedUnits.forEach(function(unit) {
-        var machine = this._createMachine();
+        // Get the application name from the unit id then grab the series
+        // for that application.
+        var series = db.services.getById(unit.id.split('/')[0]).get('series');
```

Comment:
It does but the service name and id are not the same thing at this point - they are only the same once deployed (yay legacy code)